### ISSUE TITLE
TinyMCE

### ISSUE THREAD

**Issue by szepeviktor** [xwp/wp-shortcode-ui-richtext#2]
Would it be possible to use TinyMCE as HTML editor?

**Comment by mehigh** [comment]
Hello Viktor,

I don't believe there is any reason stopping this from being a possibility.

The reason I looked for an alternatives are multiple:
- TinyMCE offers everything and the kitchen sync, and deactivating features would require development time
- The UI of TinyMCE isn't the friendliest of them all
- No experience had with its API

Off the top of my head, in order to swap out SummerNote by TinyMCE, a start would be making use of the basic example from here: https://www.tinymce.com/docs/demo/basic-example/

And replace 

``` js
$( selector ).summernote();
```

by the TinyMCE [init functions](https://www.tinymce.com/docs/api/tinymce/tinymce.editor/#init).

and

``` js
.summernote( 'code' ) )
```

by the TinyMCE [getContent function](https://www.tinymce.com/docs/api/tinymce/tinymce.editor/#getcontent)

and

``` js
.summernote( 'destroy' )
```

with [the destroy API call](https://www.tinymce.com/docs/api/tinymce/tinymce.editor/#destroy) in the following file:
https://github.com/xwp/wp-shortcode-ui-richtext/blob/master/js/richtext.js

Viktor, (or the world), if you're looking into making it a feature, since I might not dive into this too quickly, do open up a PR, and I'll update the plug-in to make summernote optional with TinyMCE becoming the default editor. 

With my suggestions, do you think you can make any progress towards achieving this?

**Comment by szepeviktor** [comment]
I think there must be a WP way to activate TinyMCE on a textarea.

**Comment by soderlind** [comment]
@szepeviktor There is, or at least it was when I looked at it a long time ago: https://soderlind.no/front-end-editor-in-wordpress-3-3/

**Comment by mehigh** [comment]
Well there's an open ticket about the fact that one can't initialise TinyMCE through an API:
https://core.trac.wordpress.org/ticket/35760
So maybe when that's tackled, we can add it to this plug-in.

**Comment by robrecord** [comment]
Would really like this feature, because it would save my client having to learn two interfaces.

**Comment by szepeviktor** [comment]
@robrecord Have you seen Shortcake?

**Comment by robrecord** [comment]
@szepeviktor Yes I've been using it, hence why I've been trying out this extension plugin. Is there something new that adds this feature?

**Comment by szepeviktor** [comment]
> @robrecord Have you seen Shortcake?

Excuse me. This was an inappropriate question.

**Comment by tillkruss** [comment]
I'd love to see TinyMCE as default as well.

**Comment by mehigh** [comment]
There is some active development on
https://core.trac.wordpress.org/ticket/35760
which would allow the dynamic instantiation of TinyMCE via JS, just like the support from SummerCode editor I used here. 
if that lands in a WP release soon, the plug-in could be upgraded to support TinyMCE.

**Comment by mehigh** [comment]
Merged #9 to master, will be released with 0.3.
